klpbuild/klplib/kernel_tree.py

```diff
@@ -78,6 +78,58 @@ def get_commit_data(commit, savedir=None):
     return date, title
 
 
+@__check_kernel_tags_are_fetched
+def get_commit_body(commit, file_path):
+    """
+    Get the changes done to the given file in a specific commit.
+    For each change, the whole function is returned as context.
+    """
+
+    kernel_tree = get_user_path("kernel_dir")
+
+    ret = subprocess.run(["git", "-C", kernel_tree, "show", "-W",
+                          "--pretty='%b'", f"{commit}", f"{file_path}"],
+                         check=False, capture_output=True, text=True)
+    return ret.stdout
+
+
+@__check_kernel_tags_are_fetched
+def find_commit(subject, branch):
+    """
```

Comment:
This was already tried in the past, and had a couple of corner cases. I don't recall anymore, since it was a long time ago, but let's see how this one behaves.

Reply:
Ah! You're using `kernel-sources` and git-show with `-W`, so yeah, this might work!!!